aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java

```diff
@@ -116,13 +116,18 @@ public void convert(final Map<String, File> contentPackages) throws IOException
                             .setEmitter(DefaultPackagesEventsEmitter.open(this.featureOutputDirectory))
                             .setResourceFilter(getResourceFilter());
 
-        for(Map.Entry<String, File> entry : contentPackages.entrySet()) {
-            logger.info("Converting package {}", entry.getKey());
-            try {
-                converter.convert(entry.getValue());
-            } catch (final Throwable t) {
-                throw new IOException("Content Package Converter Exception " + t.getMessage(), t);        
+        try {
+            for(Map.Entry<String, File> entry : contentPackages.entrySet()) {
+                logger.info("Converting package {}", entry.getKey());
+                try {
+                    converter.convert(entry.getValue());
+                } catch (final Throwable t) {
+                    throw new IOException("Content Package Converter Exception " + t.getMessage(), t);        
+                }
             }
+        } finally {
+            // make sure to remove the temp folders
+            converter.cleanup();
```

Comment:
If the assumption is that the temp folders can always be deleted, it should be treated as an error if that fails. Therefore, I propose to let `cleanup` throw `IOException`, that can be handled (or bubbled through to the calling code).

It also goes in line with the suggestion of making AemPackageConverter Closeable, as it is able to throw `IOException`.

Reply:
https://github.com/apache/sling-org-apache-sling-feature-cpconverter/pull/86